compiler/damlc/tests/daml-test-files/LfStableContractKeys.daml

```diff
@@ -67,96 +71,120 @@ template MaintainerNotSignatory
-    lookupByKey @TextKey (alice, "blah")
-  assertMsg "None == noCid1" (None == noCid1)
+    createAndExerciseCmd (Helper alice) (LookupByKey (alice, "blah"))
+  assertMsg "None == noCid1" (isNone noCid1)
 
   noCid2 <- submit bob do
-    lookupByKey @TextKey (bob, "some-key")
-  assertMsg "None == noCid2" (None == noCid2)
+    createAndExerciseCmd (Helper bob) (LookupByKey (bob, "some-key"))
+  assertMsg "None == noCid2" (isNone noCid2)
 
   submit alice do
-    exercise cid1 TextKeyChoice
+    exerciseCmd cid1 TextKeyChoice
 
   noCid3 <- submit alice do
-    lookupByKey @TextKey (alice, "some-key")
-  assertMsg "None == noCid3" (None == noCid3)
+    createAndExerciseCmd (Helper alice) (LookupByKey (alice, "some-key"))
+  assertMsg "None == noCid3" (isNone noCid3)
 
   -- if we consume an absolute CID within a single
   -- submission, we witness that it does not exist
   cid4 <- submit alice do
-    create TextKey with
+    createCmd TextKey with
       p = alice
       k = "some-key-2"
       disclosedTo = [bob]
   noCid4 <- submit alice do
-    exercise cid4 TextKeyChoice
-    lookupByKey @TextKey (alice, "some-key-2")
-  assertMsg "None == noCid4" (None == noCid4)
+    exerciseCmd cid4 TextKeyChoice
+      *> createAndExerciseCmd (Helper alice) (LookupByKey (alice, "some-key-2"))
+  assertMsg "None == noCid4" (isNone noCid4)
 
-  -- we should be able to wintness keys within a single submit
-  submit alice do
-    cid <- create TextKey with
+  -- we should be able to witness keys within a single submit
+  (cid, mbCid) <- submit alice do
+    cid <- createCmd TextKey with
       p = alice
       k = "same-submit-key"
       disclosedTo = []
-    mbCid <- lookupByKey @TextKey (alice, "same-submit-key")
-    assertMsg ("Some cid == mbCid") (Some cid == mbCid)
+    mbCid <- createAndExerciseCmd (Helper alice) (LookupByKey (alice, "same-submit-key"))
+    pure (cid, mbCid)
+  assertMsg ("Some cid == mbCid") (Some cid == mbCid)
+
   -- and a single choice
   createAndLookupCid <- submit alice do
-    create CreateAndLookup with
+    createCmd CreateAndLookup with
       p = alice
       k = "same-choice-key"
   submit alice do
-    exercise createAndLookupCid Good
+    exerciseCmd createAndLookupCid Good
   -- creating a dupe in a single choice should fail
   submitMustFail alice do
-    exercise createAndLookupCid Dupe
+    exerciseCmd createAndLookupCid Dupe
 
   -- nonconsuming choices should not affect key activeness
-  submit alice $ do
-    cid <- create TextKey with
+  cid <- submit alice do
+    createCmd TextKey with
       p = alice
       k = "non-consuming-choice"
       disclosedTo = []
-    exercise cid TextKeyNonconsumingChoice
-    mbCid <- lookupByKey @TextKey (alice, "non-consuming-choice")
-    assertMsg ("Some cid == mbCid") (Some cid == mbCid)
+  submit alice do
+    exerciseCmd cid TextKeyNonconsumingChoice
+  mbCid <- submit alice do
+    createAndExerciseCmd (Helper alice) (LookupByKey (alice, "non-consuming-choice"))
```

Comment:
Could this be one transaction?

Reply:
Oh - i see not all because we need that `cid` later
if only there was a createAndExercise that gave back the cid as well (we can add that :) )

Reply:
that would be nice, though the deeper issue here is that `Commands` is `Applicative` but not ~~`Monad`~~ _`Action`_

Reply:
Yes ofc, and it must be.
I think you could at least combine the last 2 submits though.
I do want to introduce that version of createAndExercise in daml3-script, as its trivial with the new implementation

Reply:
ah yes that should let us bring it down to 2 submits

Reply:
Did we want to make this change?

Reply:
I assume not from the resolved, I'll just approve